src/components/ProfileForm/index.tsx

```diff
@@ -1,33 +1,109 @@
-import Input from '../styled-components/Input';
+import { useState, SyntheticEvent, useCallback, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import LabelBasicInput from '../LabelBasicInput';
+import LabelBasicSelect from '../LabelBasicSelect';
 import Button from '../styled-components/Button';
-import Label from '../styled-components/Label';
-import './style.scss';
+import API from '../../API/API';
+import styles from './style.module.scss';
+
+const mbtiList: Array<string> = [
+  'ESTP',
+  'ESFP',
+  'ENFP',
+  'ENTP',
+  'ESTJ',
+  'ESFJ',
+  'ENFJ',
+  'ENTJ',
+  'ISTJ',
+  'ISFJ',
+  'INFJ',
+  'INTJ',
+  'ISTP',
+  'ISFP',
+  'INFP',
+  'INTP',
+];
+
+const genderList: Array<string> = ['남', '여'];
 
 const ProfileForm = () => {
-  const onClickCancel = () => {
+  const { userId } = useParams<string>();
+  const [userNickname, setUserNickname] = useState<string>('');
+  const [userEmail, setUserEmail] = useState<string>('');
+  const [userMbti, setUserMbti] = useState<string>('');
+  const [userGender, setUserGender] = useState<string>('');
+
+  const onClickCancel = useCallback((e: SyntheticEvent) => {
+    e.preventDefault();
     window.location.href = '/';
+  }, []);
+
+  const fetchProfile = async () => {
+    const data = {
+      userId,
+      nickname: userNickname,
+      email: userEmail,
+      mbti: userMbti,
+      gender: userGender,
+    };
+    await API.updateProfile(data);
+  };
+
+  const onSubmitForm = (e: SyntheticEvent) => {
+    e.preventDefault();
+    fetchProfile();
   };
 
+  const onChangeNicknameEvent = useCallback((e: SyntheticEvent<HTMLInputElement>) => {
+    setUserNickname(e.currentTarget.value);
+  }, []);
+
+  const onChangeEmailEvent = useCallback((e: SyntheticEvent<HTMLInputElement>) => {
+    setUserEmail(e.currentTarget.value);
+  }, []);
+
+  const onChangeMbtiEvent = useCallback((e: SyntheticEvent<HTMLSelectElement>) => {
+    setUserMbti(e.currentTarget.value);
+  }, []);
+
+  const onChangeGenderEvent = useCallback((e: SyntheticEvent<HTMLSelectElement>) => {
+    setUserGender(e.currentTarget.value);
+  }, []);
+
+  useEffect(() => {
+    const getProfile = async (): Promise<void> => {
+      const { data } = await API.getId({ userId });
```

Comment:
클라이언트에서 request로 넘겨주는 값은 data일 수 있는데

api호출 후 받는 값은 response이므로 data 대신 response로 적는것은 어떨까요?
```suggestion
      const response = await API.getId({ userId });
```

Reply:
response로 수정하는것이 좋을 것 같습니다